impl-base/src/main/java/org/jboss/shrinkwrap/impl/base/URLPackageScanner.java

```diff
@@ -156,7 +156,8 @@ private void handle(File file, String packageName) throws ClassNotFoundException
                 String className = packagePrefix + child.getName().substring(0, child.getName().lastIndexOf(SUFFIX_CLASS));
                 foundClass(className, prefix + className.replace( '.', '/' ) + SUFFIX_CLASS );
             } else if (child.isDirectory() && addRecursively) {
-                handle(child, packageName + "." + child.getName());
+                final String nextPackageName = packageName.length() > 0 ? packageName + "." + child.getName() : child.getName();
```

Comment:
I would rather assume following behaviour to be in place:

  * passing `""` would mean we include `default` package (and that counts for `.addPackage(String)` too)
  * passing `"*"` to `.addPackages()` would result in including everything

Would it make sense to improve it in such a way?

Summoning @ALRubinger ?